icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -1641,6 +1641,25 @@ class RemapRule : public SegmentationRule {
         UnicodeString result;
         int i = 0;
         int offset = 0;
+        // We find all matches of the `pattern_` and replace them according to
+        // the `replacement_`, producing the new remapped string `result`.
+        // For every position i in the original string,
+        // `resolved[i].indexInRemapped` is nullopt if i lies within a replaced
+        // match, and is set to the new index in `result` otherwise, by adding
+        // the accumulated difference `offset` between match lengths and
+        // replacement lengths.
+        // Consider a 4-codepoint, 6 code unit string s = ⟨ 𒀀, ◌́, ␠, ◌𝅲 ⟩, where
+        // ␠ stands for U+0020 and U+12000 𒀀 and U+1D172 ◌𝅲 each require two code
+        // units, and apply the following two rules:
+        //   1. (?<X>\P{lb=SP}) \p{lb=CM}* → ${X}
+        //   2. \p{lb=CM} → A
+        // The string remapped and the indexInRemapped values change as follows:
+        // indexInRemapped                   remapped string  rule        final
+        // (aligned on the initial string)                    applied     offset
+        //    𒀀     ◌́     ␠     ◌𝅲
+        // 0  1  2     3     4  5  6         ⟨ 𒀀, ◌́, ␠, ◌𝅲 ⟩   (none)
+        // 0  ␀  ␀     2     3  4  5         ⟨ 𒀀,    ␠, ◌𝅲 ⟩   1                -1
+        // 0  ␀  ␀     2     3  ␀  4         ⟨ 𒀀,    ␠, A ⟩   2                -1
```

Comment:
should the final offset be -2?

Reply:
No, the `offset` variable is with respect to the previous value of indexInRemapped, so 6 + -1 + -1 = 4.